refactor: remove sqlalchemy dependency (#20)
gabor-boros/questdb-statuspage: README.md

Let's recap on our code above, starting from the top:

* we added `defaultdict` import (we'll explain that later)
* extended the function decorator to use `response_model=List[SignalResponse]`, the response model we defined already
* changed the function signature to include a `limit` parameter and set its default value to `60` since we will monitor HTTP status every minute

> * select the records from the database and prepare a dictionary for the parsed `Signal`s.

Reviewer: Is there a particular reason why select is present tense while the previous points are in past tense? Also note that this bullet point has a full stop.

Author: No reason, it was an overlook. How about this one?

> selected the records from the database and prepared a dictionary for the parsed `Signal`s.

@celery_app.task
def monitor():
    # ...

    query = f"""
    INSERT INTO signals(received,url,http_status,available)
    VALUES(systimestamp(), '{signal.url}', {signal.http_status}, {signal.available});
    """

    with pool.connection() as conn:
        conn.execute(query)
```

Congratulations! You just arrived at the last part of the backend service implementation. We did many things and built a service that can periodically check the website's status, save it in the database, and expose the results through an API.

The very last thing we need to address is to allow connections initiated by the frontend later on. As it will run on localhost:3000 and we don't use domain names, the port is different hence all requests will be rejected with errors related to [Cross-Origin Resource Sharing](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS).

> For the sake of simplicity, we are going to allow all origins. This is highly not recommended in production, but for the tutorial, it will be sufficient for us.

Reviewer: Highly not recommended feels weird. How about:

```suggestion
For the sake of simplicity, we are going to allow all origins. We highly recommended to avoid this in production, but for the tutorial, it will be sufficient for us.

```
Also, should we elaborate why it is not recommended, or what kind of stuff should be defined in production?

Author: Absolutely make sense! Thank you for the recommendation.

> Also, should we elaborate why it is not recommended, or what kind of stuff should be defined in production?

Actually, the reason it is not recommended originates from attacks that could be performed when the CORS settings are using `*` for origins. Since it is a tutorial, it does not really matter since it won't run in production as is.

Reviewer: I am not technical enough to decide but maybe add this as a reminder to part of the sentence? Something like:

"We highly recommended to avoid this in production for security reason but for the tutorial, it will be sufficient for us."

I just find myself asking this question when reading it but I trust your thought on this :)
